src/twisted/internet/defer.py

```diff
@@ -515,54 +542,151 @@ def addCallbacks(
         #     the caller should treat the result as the new type, consistently.
         return cast(Deferred[_NextDeferredResultT], self)
 
+    @overload
+    def addCallback(
+        self,
+        callback: Callable[Concatenate[_DeferredResultT, _P], Failure],
+        *args: _P.args,
+        **kwargs: _P.kwargs,
+    ) -> Deferred[_NextDeferredResultT]:
+        ...
+
+    @overload
     def addCallback(
         self,
         callback: Callable[
-            ...,
-            "Union[_NextDeferredResultT, Deferred[_NextDeferredResultT]]",
+            Concatenate[_DeferredResultT, _P], Deferred[_NextDeferredResultT]
         ],
-        *args: object,
-        **kwargs: object,
-    ) -> "Deferred[_NextDeferredResultT]":
+        *args: _P.args,
+        **kwargs: _P.kwargs,
+    ) -> Deferred[_NextDeferredResultT]:
+        ...
+
+    @overload
+    def addCallback(
+        self,
+        callback: Callable[Concatenate[_DeferredResultT, _P], _NextDeferredResultT],
+        *args: _P.args,
+        **kwargs: _P.kwargs,
+    ) -> Deferred[_NextDeferredResultT]:
+        ...
+
+    def addCallback(
+        self,
+        callback: Any,
+        *args: Any,
+        **kwargs: Any,
+    ) -> "Deferred[Any]":
         """
         Convenience method for adding just a callback.
 
         See L{addCallbacks}.
         """
         return self.addCallbacks(callback, callbackArgs=args, callbackKeywords=kwargs)
 
+    @overload
     def addErrback(
         self,
-        errback: Callable[
-            ...,
-            "Union[Failure, _NextDeferredResultT, Deferred[_NextDeferredResultT]]",
-        ],
-        *args: object,
-        **kwargs: object,
+        errback: Callable[Concatenate[Failure, _P], Deferred[_NextDeferredResultT]],
+        *args: _P.args,
+        **kwargs: _P.kwargs,
+    ) -> "Deferred[Union[_DeferredResultT, _NextDeferredResultT]]":
+        ...
+
+    @overload
+    def addErrback(
+        self,
+        errback: Callable[Concatenate[Failure, _P], Failure],
+        *args: _P.args,
+        **kwargs: _P.kwargs,
+    ) -> "Deferred[Union[_DeferredResultT]]":
+        ...
+
+    @overload
+    def addErrback(
+        self,
+        errback: Callable[Concatenate[Failure, _P], _NextDeferredResultT],
+        *args: _P.args,
+        **kwargs: _P.kwargs,
+    ) -> "Deferred[Union[_DeferredResultT, _NextDeferredResultT]]":
+        ...
+
+    def addErrback(
+        self,
+        errback: Callable[..., Any],
```

Comment:
If we have to annotate this, it seems desirable for the annotation of `errback` to `addErrback` match the annotation of `callable` to `addCallback`.

Reply:
Yes, good catch.